catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/PollerRunner.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.util.impl;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import com.google.common.util.concurrent.MoreExecutors;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link #setPollIntervalMinutes} starts the process to periodically update the {@link Poller}.
+ *
+ * @param <K> the key type of the {@link Poller}
+ * @param <V> the value type of the {@link Poller}
+ */
+abstract class PollerRunner<K, V> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PollerRunner.class);
+
+  private static final long MINIMUM_POLL_INTERVAL_MINUTES = 1;
+
+  private final Poller<K, V> poller;
+
+  private final ScheduledExecutorService scheduledExecutorService;
+
+  private volatile long pollIntervalMinutes;
+
+  @Nullable private volatile Future scheduledPollingFuture;
+
+  /**
+   * {@link #setPollIntervalMinutes} must be called to start the polling
+   *
+   * @throws IllegalArgumentException if {@code initialPollIntervalMinutes} is less than {@value
+   *     #MINIMUM_POLL_INTERVAL_MINUTES}
+   * @throws NullPointerException if any of the parameters are {@code null}
+   * @throws IllegalArgumentException if the {@code scheduledExecutorService} {@link
+   *     ScheduledExecutorService#isShutdown()}
+   */
+  PollerRunner(
+      final Poller<K, V> poller,
+      final long initialPollIntervalMinutes,
+      final ScheduledExecutorService scheduledExecutorService) {
+    this.poller = notNull(poller);
+    if (scheduledExecutorService.isShutdown()) {
+      throw new IllegalArgumentException("scheduledExecutorService argument may not be shutdown");
```

Comment:
❓ Did you mean `("scheduledExecutorService argument may be shutdown")` ?

Reply:
See https://github.com/codice/ddf/pull/3858#discussion_r248775239